trench/backends/base.py

```diff
@@ -73,12 +78,18 @@ def validate_confirmation_code(self, code: str) -> bool:
         return self.validate_code(code)
 
     def validate_code(self, code: str) -> bool:
-        return self._get_otp().verify(otp=code)
+        if self._mfa_method.is_totp:
+            return self._get_otp().verify(otp=code)
+        else:
+            return self._get_otp().verify(otp=code, counter=self._mfa_method.counter)
```

Comment:
Perhaps a successful verify should also increase the counter such that the same code cannot be used again.

Reply:
Or if using a timestamp, set that timestamp to `None` essentially marking the same code as invalid.